pandas/core/reshape/melt.py

```diff
@@ -72,7 +72,14 @@ def melt(
                     "The following 'value_vars' are not present in "
                     f"the DataFrame: {list(missing)}"
                 )
-        frame = frame.loc[:, id_vars + value_vars]
+        # use `iloc` instead of `loc` when `col_level` is specified
+        if col_level is not None:
```

Comment:
you can you use `.get_indexer` in both cases and *always* use `.iloc`?

Reply:
yes, better to be consistent. Done!